### DISCUSSION TITLE
Running Locally

### DISCUSSION THREAD

**Discussion by ih4ck3r** [pwillworth/dfkreport#15] [Q&A]
I know, this has been asked already but I didn't find an answer there. I'm interested on running this locally as well. Can you give a brief idea on how to? I've cloned the repo, installed the requirements, setup db, and served using "python3 -m http.server 8000", but it just display the python script on the brower. Not asking for a step by step tutorial, just a quick know how so I know i'm heading the right direction. Got 0 knowledge on python. Thanks

**Comment by pwillworth** [comment]
I have never tried running the site under python's http server module, so have no idea how/if that would work.  The live site runs under Apache, so you might want to setup Apache locally to get started and configure it to execute .py files.  Once you have apache serving up the pages, you need to run the process ./main.py to watch for and process report requests.

**Reply by ih4ck3r** [reply]
Tried it on apache2, it just displays the python script so i tried searching and did several solution like using cgi, wsgi mod and passenger but it just gives me random errors. The .py files seems to be executing ok but the template part (jinja2) just won't render the template on browser. Can you give me an idea to make that working? jinja2 is already installed.

**Reply by pwillworth** [reply]
You might find the setup information in the readme on one of my other projects helpful, it has a similar setup and dependencies: [https://github.com/pwillworth/galaxyharvester#readme](url)

**Reply by ih4ck3r** [reply]
Thanks for the info. Finally got it somewhat working. What do I need to add on process for reports requests? main.py? i tried it adding it on cron but it is requiring wallet address and stuff?

**Reply by pwillworth** [reply]
No, you want to run watch.py.  Just keep that running and it kicks off main.py when it gets a report request.

**Reply by ih4ck3r** [reply]
thanks for the help. all works now. i also have to change the endpoint on dfk-chain (main) to 1.1, 1.0 doesn't seem to work anymore.
